### ISSUE TITLE
Abstract method parameter default values do not get nullability warnings

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#48847]
**Version Used**: d5931e08dd388356176821aa25df094d0e9e94a8

**Steps to Reproduce**:

```cs
#nullable enable

public abstract class C
{
    public abstract void M1(string s = null);
}

public delegate void Del1(string s = null);
```

[SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4GIB2BXANliwtwAEcaeBAsAFBUACAzIQCZwEDmE8hNALIQCIsAPABUAfAAphhGIQC8TOADMI2GAEoA3FVoMaAJkIBhEaKoBvKoStddvALIAGSdLWEzhAL7bK1wgHo/QgAhOAALCAA3AEsAewRCRTjCMDgwYERkuBhQmMYoABoEpIBjGLAABwgEKKgYtEtrei57AEZnGXlmZVU1BqsLH19rR3F1DX9AgHcqtCi0ViMAZQAOADYHbhBCAAUYqCgo/CJMHEIEJUQSYqIq1gwUtBg+z2fnpp5COz1xXsH+598AiwJgUWDk0jkok+TjGE2IAA9ynBivBGIRpghZvNCsAMDJGFFGGgAOQyVhZQh1OAA6xAkHMLAtcEdKHuDoKbpYGDjEawjzjALomZzBaGFbrTY7PYHI6EE5g86KS5oa6EW73EhPP4vSgeIA=)

**Expected Behavior**: Warnings on the parameter default values for `M1` and `Del1`.

**Actual Behavior**: No warnings.

Basically, it seems that when the method has no body, we entirely fail to analyze its parameter default values. It suggests that the approach to synthesize the parameter default values and pretend to assign them in NullableWalker is not quite right. When a parameter default value is included in syntax, it may be associated directly with a varying number of method bodies:

- 0 for delegates and abstract methods
- 1 for ordinary methods
- 2 for indexers with get and set accessors

edit: Expected behavior of this scenario has changed based on the proposal [nullable-parameter-default-value-analysis.md](https://github.com/dotnet/csharplang/blob/master/proposals/nullable-parameter-default-value-analysis.md)
